src/types/User.ts

```diff
@@ -90,23 +106,29 @@ export const UserMutation = extendType({
       type: CreateUserMutationResult,
       description: 'Register a new user account',
       args: {
-        name: stringArg({ description: 'Name of the user' }),
-        username: nonNull(stringArg({ description: 'Username for user' })),
-        password: nonNull(stringArg({ description: 'Password for user' })),
-        role: nonNull(Role),
+        input: arg({ type: nonNull(CreateUserInput) }),
       },
-      resolve: async (_, args, ctx) => {
-        const { name, username, password, role } = args;
-        const { prisma } = ctx;
-        const newUser = await (prisma as PrismaClient).user.create({
+      resolve: async (_, args, { prisma }) => {
+        const { name, username, password, role } = args.input;
+        const userExists = await prisma.user.findFirst({ where: { username } });
+        if (userExists) {
+          return {
+            success: false,
+            message: 'User already exists',
+          };
+        }
+
+        const hashPwd = await generateHashPassword(password);
+
+        const newUser = await prisma.user.create({
           data: {
-            id: username.toLowerCase().replace(' ', '-'),
             name,
             username,
-            password,
+            password: hashPwd,
             role,
           },
         });
+
         if (!newUser) return { success: false, message: 'Could not create user' };
         else return { success: true, message: 'User created' };
```

Comment:
don't need an else here
```suggestion
        return { success: true, message: 'User created' };
```